tools/eslint-rules/rules/consistent-callback-type.spec.ts

```diff
@@ -3,6 +3,10 @@ import { rule, RULE_NAME } from './consistent-callback-type';
 
 const ruleTester = new RuleTester();
 
+global.structuredClone = jest.fn(val => {
```

Comment:
started to fail after change in this version: `@typescript-eslint/utils@8.5.0`.

`JSON.parse(JSON.stringify(val))` was replaced with `structuredClone`. 

![Screenshot 2024-10-15 at 16 41 20](https://github.com/user-attachments/assets/2a76ef9e-243a-46e2-9d76-e8d7101916e3)
![Screenshot 2024-10-15 at 16 45 29](https://github.com/user-attachments/assets/61067812-6ec8-4e4b-96cf-b64f2a5eb2ee)

Node supports `strucutredClone` since v17. This looks odd, added for failing tests in order pipeline is passing. [Found also this](https://github.com/jsdom/jsdom/issues/3363#issuecomment-1467894943), which might be the better way of fixing this.